Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/ActionManager.cpp

```diff
@@ -6,12 +6,99 @@
  *
  */
 
-#include <AzToolsFramework/ActionManager/Action/ActionManager.h>
+#include <QApplication>
+#include <QtGui/private/qguiapplication_p.h>
+#include <QShortcutEvent>
+
+#include <AzQtComponents/Components/WindowDecorationWrapper.h>
 
+#include <AzToolsFramework/ActionManager/Action/ActionManager.h>
 #include <AzToolsFramework/ActionManager/Menu/MenuManagerInterface.h>
 
 namespace AzToolsFramework
 {
+    //! This class is used to install an event filter on each widget that the action contexts are registered to in order to properly handle
+    //! ambiguous shorcuts.
+    class ActionContextWidgetWatcher : public QObject
+    {
+    public:
+        explicit ActionContextWidgetWatcher(QObject* parent)
+            : QObject(parent)
+        {
+        }
+
+        bool eventFilter(QObject* watched, QEvent* event) override
+        {
+            switch (event->type())
+            {
+            case QEvent::Shortcut:
+            {
+                QShortcutEvent* shortcutEvent = static_cast<QShortcutEvent*>(event);
```

Comment:
uber nit: With `static_cast` as the type is already on the right of the expression we can just use `auto` on the left to reduce redundancy of the type info (repeated below)

Reply:
Good call, will fix!